cachix/src/Cachix/Client/Commands.hs

```diff
@@ -287,7 +293,10 @@ pushStorePath env name storePath = do
 
 -- Utils
 
--- TODO: should one error abort the whole pushing process? Or do a retry? Or keep going and then fail?
+-- Retry up to 5 times for each store path
+retryPath :: (RetryStatus -> IO a) -> IO a
+retryPath = recoverAll def
```

Comment:
`recoverAll` is too broad. We don't want to retry on Ctrl+C for example. Use something like [`isSyncException`](https://hackage.haskell.org/package/safe-exceptions-0.1.7.0/docs/Control-Exception-Safe.html#v:isSyncException) with `Control.Retry.recovering`

Reply:
Ok

Reply:
Needs an inline comment about skipping asyncs